android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +84,84 @@ public void init(String siteKey, String touchpointUri, String apiKey,
-                  "\"optimizations: " + oneResponse.getOptimizations().toString() + ", " +
-                  "\"statusCode\": " + oneResponse.getHttpStatusCode() + ", " +
-                  "\"tid\": \"" + oneResponse.getTid() + "\", " +
-                  "\"trackers\": " + oneResponse.getTrackers().toString() + "}"
-          );
-        }
-
-        @Override
-        public void onError(@NotNull OneSDKError oneSDKError) {
-          notifyProblem(promise, "" + oneSDKError.getSystemCode(), oneSDKError.getErrorMessage());
-        }
-
-        @Override
-        public void onFailure(@NotNull OneAPIError oneAPIError) {
-          notifyProblem(promise, "" + oneAPIError.getHttpStatusCode(), oneAPIError.getErrorMessage());
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
+        } else {
+          response = One.sendInteractionLegacySupport(true, sendInteractionRequest).join();
         }
-      });
-    }
+        One.processResponse(response);
+        WritableNativeMap responseMap = responseObjectToReadableMap(response);
+        notifyResult(promise, responseMap);
+      } catch (ExecutionException error) {
+        notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Completion Error: " + error.getCause());
+      } catch (OneSDKError error) {
+        notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction SDK Error: " + error.getErrorMessage());
+      } catch (OneAPIError error) {
+        notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Api Error: " + error.getErrorMessage());
+      }
+    });
   }
 
   @ReactMethod
-  public void sendProperties(String interaction, ReadableMap propertiesMap) {
+  public void sendProperties(String interactionPath, ReadableMap propertiesMap) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
-    final OneRequest sendPropertiesRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+    final OneRequest sendInteractionRequest = new OneRequest.Builder()
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-
-    final OneCall sendPropertiesCall = One.sendProperties(sendPropertiesRequest);
-    sendPropertiesCall.enqueue(null);
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
```

Comment:
you need to call send properties, not send interaction.

Reply:
updated